packages/@lwc/engine-core/src/framework/vm.ts

```diff
@@ -280,7 +286,11 @@ function resetComponentStateWhenRemoved(vm: VM) {
 // old vnode.children is removed from the DOM.
 export function removeVM(vm: VM) {
     if (process.env.NODE_ENV !== 'production') {
-        if (!lwcRuntimeFlags.ENABLE_NATIVE_CUSTOM_ELEMENT_LIFECYCLE) {
+        if (
+            !shouldUseNativeCustomElementLifecycle(
```

Comment:
@nolanlawson do you think it would it be worth it to cache the value on VM?

Reply:
I don't think so, so. I don't think computing this should be very expensive – it's just about comparing integer values mostly.